packages/starlight/components/MarkdownContent.astro

```diff
@@ -95,6 +95,11 @@
   }
 
   .content :global(table) {
+    display: block;
+    width: 100%;
+    width: max-content;
+    max-width: 100%;
+    overflow: auto;
     border: 1px solid var(--sl-color-gray-5);
```

Comment:
Thanks @rviscomi!

I think this can be even simpler! If we remove the border from the table, we get the effect we want:

```suggestion
    display: block;
    overflow: auto;
```

This is also maybe the preferable fix because setting `display: block` disabled collapsing the table’s border with cell borders so we’re seeing a double border currently. This branch on the left, live site on the right:

<img width="935" alt="Screenshot of two tables, one with a 2 pixel border, one with 1 pixel as desired" src="https://github.com/withastro/starlight/assets/357379/7ea7379b-9799-4ea6-8220-08edd1019599">